internal/controller/backend_security_policy.go

```diff
@@ -89,6 +90,28 @@ func (c *BackendSecurityPolicyController) reconcile(ctx context.Context, bsp *ai
 		if err != nil {
 			return res, err
 		}
+		secretName, useGeneratedSecret := getSecretNameAndOwnership(bsp)
+		if useGeneratedSecret {
+			var secret *corev1.Secret
+			secret, err = rotators.LookupSecret(ctx, c.client, bsp.Namespace, secretName)
+			if err != nil {
+				return res, fmt.Errorf("failed to lookup backend security policy secret %s/%s: %w",
```

Comment:
This is also a bug after successfully rotated the credential

```suggestion
				panic(fmt.Sprintf("BUG: failed to lookup backend security policy secret %s/%s: %w",bsp.Namespace, secretName, err))
```

Reply:
done